apps/heureka/src/hooks/useCommunication.js

```diff
@@ -3,65 +3,103 @@
 import {
   useUserActivityActions,
-  useAuthAppLoaded,
-  useAuthIsProcessing,
-  useAuthError,
-  useAuthLoggedIn,
-  useAuthLastAction,
   useAuthActions,
+  useIssueMatchesActiveFilters,
+  useServiceActiveFilters,
+  useComponentActiveFilters,
+  useFilterActions,
+  useIssueMatchesFilterLabels,
+  useServiceFilterLabels,
+  useComponentFilterLabels,
 } from "./useAppStore"
-import { AUTH_ACTIONS } from "../lib/slices/createAuthDataSlice"
 
 const useCommunication = () => {
+  console.debug("[heureka] useCommunication setup")
   const { setIsActive } = useUserActivityActions()
-  const authAppLoaded = useAuthAppLoaded()
-  const authIsProcessing = useAuthIsProcessing()
-  const authError = useAuthError()
-  const authLoggedIn = useAuthLoggedIn()
-  const authLastAction = useAuthLastAction()
-  const { setData: authSetData, setAppLoaded: authSetAppLoaded } = useAuthActions()
+  const { setData: authSetData } = useAuthActions()
+
+  // Active filters for each entity
+  const issueMatchesActiveFilters = useIssueMatchesActiveFilters()
+  const serviceActiveFilters = useServiceActiveFilters()
+  const componentActiveFilters = useComponentActiveFilters()
+
+  // Filter labels for each entity
+  const issueMatchesFilterLabels = useIssueMatchesFilterLabels()
+  const serviceFilterLabels = useServiceFilterLabels()
+  const componentFilterLabels = useComponentFilterLabels()
+
+  const { setActiveFilters } = useFilterActions()
+
+  const setAuthData = useCallback(
+    (data) => {
+      if (!data) return
+
+      // Set the auth data
+      authSetData(data)
+
+      // The following code exists for historical reasons and should be refactored
+      // We preset the support group filter based on auth data. This should be done
+      // with predefined filters prop
+
+      // Handle support group filters for IssueMatches
+      if (
+        !issueMatchesActiveFilters?.support_group &&
+        data?.auth?.parsed?.supportGroups &&
+        issueMatchesFilterLabels?.includes("support_group")
+      ) {
+        setActiveFilters("IssueMatches", { support_group: data.auth.parsed.supportGroups })
+      }
+
+      // Handle support group filters for Services
+      if (
+        !serviceActiveFilters?.support_group &&
+        data?.auth?.parsed?.supportGroups &&
+        serviceFilterLabels?.includes("support_group")
+      ) {
+        setActiveFilters("Services", { support_group: data.auth.parsed.supportGroups })
+      }
+    },
+    [
+      authSetData,
+      issueMatchesFilterLabels,
+      serviceFilterLabels,
+      componentFilterLabels,
+      issueMatchesActiveFilters,
+      serviceActiveFilters,
+      componentActiveFilters,
+      setActiveFilters,
+    ]
+  )
 
   useEffect(() => {
-    // watch for user activity updates messages
+    // Watch for user activity updates
     // with the watcher we get the user activity object when this app is loaded before the Auth app
     const unwatch = watch(
```

Comment:
You probably do not need this watch, since it is not implemented yet in greenhouse. We might probably design it new with the new architecture.

Reply:
so I remove it from here then.